packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -715,7 +750,19 @@ export const MenuBarMixin = (superClass) =>
      */
     _setFocused(focused) {
       if (focused) {
-        const target = this.querySelector('[tabindex="0"]');
+        if (this.tabNavigation) {
+          // manage the submenu
+          const target = this.querySelector('[focused]');
+          const wasExpanded = this._expandedButton !== undefined && this._expandedButton !== target;
+          if (wasExpanded) {
+            this._close();
+          }
+
+          if (wasExpanded && target.item && target.item.children) {
+            this.__openSubMenu(target, true, { keepFocus: true });
+          }
```

Comment:
We could use nested `if` here as `wasExpanded` is already checked above:

```suggestion

            if (target.item && target.item.children) {
              this.__openSubMenu(target, true, { keepFocus: true });
            }
          }
```

Reply:
It has been moved according to the suggestion